src/Bcgov/DesignSystemPlugin/DocumentRepository/src/Service/AdminUIManager.php

```diff
@@ -0,0 +1,231 @@
+        DocumentMetadataManager $metadata_manager
+    ) {
+        $this->config           = $config;
+        $this->uploader         = $uploader;
+        $this->metadata_manager = $metadata_manager;
+    }
+
+    /**
+     * Add the document repository menu to the WordPress admin.
+     */
+    public function add_repository_menu(): void {
+        add_menu_page(
+            'Document Repository',
+            'Document Repository',
+            $this->config->get_capability(),
+            $this->config->get( 'menu_slug' ),
+            [ $this, 'render_repository_page' ],
+            $this->config->get( 'menu_icon' ),
+            $this->config->get( 'menu_position' )
+        );
+    }
+
+    /**
+     * Add the metadata settings submenu.
+     */
+    public function add_metadata_settings_submenu(): void {
+        add_submenu_page(
+            $this->config->get( 'menu_slug' ),
+            'Metadata Settings',
+            'Metadata Settings',
+            $this->config->get_capability(),
+            $this->config->get( 'metadata_slug' ),
+            [ $this, 'render_metadata_settings_page' ]
+        );
+    }
+
+    /**
+     * Enqueue admin scripts and styles.
+     *
+     * @param string $hook Current admin page hook.
+     */
+    public function enqueue_admin_scripts( string $hook ): void {
+        // Only load on our plugin pages.
+        $valid_hooks = [
+            'toplevel_page_' . $this->config->get( 'menu_slug' ),
+            'document-repository_page_' . $this->config->get( 'metadata_slug' ),
+        ];
+
+        if ( ! in_array( $hook, $valid_hooks, true ) ) {
+            return;
+        }
+
+        // Register and enqueue scripts.
+        $this->register_admin_scripts();
+
+        // Load the appropriate script for the current page.
+        if ( 'toplevel_page_' . $this->config->get( 'menu_slug' ) === $hook ) {
+            wp_enqueue_script( 'dswp-document-repository-app' );
+        } else {
+            wp_enqueue_script( 'dswp-document-repository-metadata-app' );
+        }
+
+        // Always load the main styles.
+        wp_enqueue_style( $this->config->get( 'css_handle' ) );
+
+        // Localize script with data.
+        $this->localize_scripts( $hook );
+    }
+
+    /**
+     * Register all admin scripts and styles.
+     */
+    private function register_admin_scripts(): void {
+        // Get the plugin root directory URL using WordPress constants.
+        $plugin_dir = WP_PLUGIN_DIR . '/design-system-wordpress-plugin';
+        $plugin_url = WP_PLUGIN_URL . '/design-system-wordpress-plugin';
+        $build_path = $plugin_dir . '/src/Bcgov/DesignSystemPlugin/DocumentRepository/build';
+
+        // Get version from file modification time, or use fallback if file doesn't exist.
+        $js_file = $build_path . '/document-repository.js';
+        $version = file_exists( $js_file ) ? filemtime( $js_file ) : time();
+
+        // Add nonce to script data.
```

Comment:
```suggestion
        // Adding global js variables like nonce, and api information.
```
**Suggestion** just including the other important data that is being added, i suspect nonce was what you started off with, but expanded